pkg/config/nodetreemodel/config.go

```diff
@@ -129,7 +148,10 @@ func (c *ntmConfig) SetWithoutSource(key string, value interface{}) {
 
 // SetDefault assigns the value to the given key using source Default
 func (c *ntmConfig) SetDefault(key string, value interface{}) {
-	c.Set(key, value, model.SourceDefault)
+	c.Lock()
+	defer c.Unlock()
+
+	_, _ = c.set(key, value, c.defaults, model.SourceDefault)
```

Comment:
Just a heads up that this will conflict with how my PR builds the default tree.

Reply:
Following the convo on your PR I think it will no longer conflict in the future.